.travis.yml

```diff
@@ -1,35 +1,63 @@
 language: rust
 
-rust:
-  - nightly
-  - beta
-  - stable
+os:
+  - linux
+
+addons:
+  apt:
+    packages:
+      # For building MUSL static builds on Linux.
+      - musl-tools
 
 matrix:
+  fast_finish: true
   allow_failures:
     - rust: nightly
-  fast_finish: true
+  include:
+    - os: linux
+      rust: nightly
+      env: TARGET=x86_64-unknown-linux-musl
+    - os: linux
+      rust: beta
+      env: TARGET=x86_64-unknown-linux-musl
+    - os: linux
+      rust: stable
+      env: TARGET=x86_64-unknown-linux-musl
+    # - os: osx
+    #   rust: nightly
+    #   env: TARGET=x86_64-apple-darwin
+    # - os: osx
+    #   rust: beta
+    #   env: TARGET=x86_64-apple-darwin
+    # - os: osx
+    #   rust: stable
+    #   env: TARGET=x86_64-apple-darwin
 
 before_script:
+  - rustup target add $TARGET
   - cargo install --force cargo-audit
   - rustup component add clippy
 
 script:
-  - cargo build
+  - cargo build --target $TARGET --release
```

Comment:
Just a note for future views here wondering why we back-flipped and are now building with `--release`:

If we do a release build again in `before_deploy` it has to compile all the deps and the project again — but will never get cached (the cache step runs after `script` and before `deploy`). So it seems logical therefore to sacrifice a small bit of speed on every build for the purposes of saving >1h compiling all the dependencies (and openssl) from scratch again on a release build.